specs/experimental/zk-fault-dipute-game/overview.md

```diff
@@ -23,4 +37,91 @@
 
 ## Proving Fault using ZKVM
 
+### Prover
+
+A prover plays a vital role in the challenge process, acting as an integral part of the challenger responsible
+for generating proof for the target block. This proof substantiates claims regarding the legitimacy of block data
+and state transitions.
+A [ZK fault proof][g-zk-fault-proof] demonstrates that a state transition from S to S’ is valid based on the
+transactions within the block. While this may seem similar to a validity proof, the key difference lies in its
+purpose. The ZK fault proof is used to demonstrate that a state transition from S to S’’ is incorrect by
+providing evidence of a valid state
+transition from S to S’.
+
+### zkVM Proof
+
+The zkVM (Zero-Knowledge Virtual Machine) is a virtual machine that executes guest program compiled with a specified
+compiler generating zero-knowledge proofs for their execution. The guest program can be written in standard programming
+languages, such as Rust and Go.
+
+#### Guest Program
+
+In the case of the ZK fault proof, the guest program is an extension of [L2 Chain Derivation][g-l2-chain-derivation]
+that includes a connectivity check among the blocks from L1 origin block and the specified block `C`. The hash of
+Block `C` is determined as the parent hash stored when the [Dispute Game][g-dispute-game] is created in Dispute Game
+Factory by calling `create()` of the Dispute Game Factory.
+If the attacker manipulates any data within the extension of L2 Chain Derivation, it will affect the block hash `C`.
+This is because all data is ultimately linked to the block hash `C` through the hash chain. Therefore if any data
+manipulating attack be thwarted by checking the block hash `C` value. For example, if an attacker creates a proof
+using a transaction that is not included in the [sequencer-batch][g-sequencer-batch], the block `C` value will change,
+preventing them from winning the challenge.
+
+#### Host Program
+
+The host program is a main part of the prover, responsible for delivering the guest program and its input to the zkVM.
+The host program first executes the guest program logic to gather the necessary data, which is equal to what is stored
+in the [PreImageOracle] during the [Optimism's Dispute Game], but the block headers from the L1 block containing
+the [sequencer-batch][g-sequencer-batch] up to block `C` must be included. Finally, the host program delivers the guest
```

Comment:
```suggestion
The host program first executes the guest program to gather the necessary data, which is called [PreImageOracle] in [Optimism's Dispute Game]. For ZKFDG, however, additional data is required for PreImageOracle, which is the L1 block data from the origin of target L2 block to block `C`. Finally, the host program delivers the guest
```

Reply:
I think we don't use PreImageOracle, so better to remove that part.
```suggestion
The host program first executes the guest program to gather the necessary data, which is same as the data stored in [PreImageOracle] during [Optimism's Fault Dispute Game]. For ZKFDG, however, additional data is required, which is the L1 block data from the origin of target L2 block to block `C`. Finally, the host program delivers the guest
```

Reply:
Actually, we use PreImageOracle off-chain (in prover), not on-chain. That's why @dongchangYoo add this part. The spec of PreImageOracle for guest program is the same with Optimism's. The only difference is the L1 Block info for validation of connectivity of L1 blocks.

Reply:
Since `PreImageOracle` is a contract, not a data, i think that
```
the necessary data, which is equal to what is stored in the [PreImageOracle] during the [Optimism's Dispute Game]
```
is more accurate than
```
necessary data, which is called [PreImageOracle] in [Optimism's Dispute Game]
```

Reply:
Oh, I see. Let's go as you write.